inc/zoo/swar/associative_iteration.h

```diff
@@ -154,6 +154,16 @@ fullAddition(SWAR<NB, B> s1, SWAR<NB, B> s2) {
     return { result, BS{carry.value()}, BS{overflow.value()} };
 };
 
+template<int NB, typename B>
+constexpr SWAR<NB, B>
+saturatingUnsignedArithmetic(SWAR<NB, B> s1, SWAR<NB, B> s2) {
+  const auto additionResult = fullAddition(s1, s2);
+  // If we carry unsigned, we need to saturate: thus OR the carry bit with the
+  // lane bits (carry because it happens to be earlier in the struct
+  // declaration)
+  return additionResult.carry.MSBtoLaneMask() | additionResult.result;
```

Comment:
Please 4 spaces indentation?
I really believe on not creating accidental levels of indentation, Kevlin Henney, etc.

Reply:
My bad, ancient habits die hard, will fix before commit.

Reply:
Will clang format be useful? I could imagine you guys might have some strong opinions about it if you haven't already added it to the repo.

Reply:
You did not watch the presentation by Tony Van Eerd I recommended you to watch, it turns out that I watched it to make sure I gave you the right reference and independently Scott watched it, a very major point in the presentation is the very answer to your question.  I will not tell you more to avoid spoilers.  Then we can discuss.